spring-xd-dirt/src/main/java/org/springframework/xd/dirt/rest/XDController.java

```diff
@@ -62,14 +64,20 @@
  * @author Gunnar Hillert
  */
 
-public abstract class XDController<D extends BaseDefinition, A extends ResourceAssemblerSupport<D, R>, R extends NamedResource> {
+public abstract class XDController<D extends BaseDefinition, A extends
+		ResourceAssemblerSupport<D, R>, R extends NamedResource, I extends BaseInstance<D>> {
 
-	private ResourceDeployer<D> deployer;
+	private AbstractInstancePersistingZKDeployer<D, I> deployer;
```

Comment:
IMO we should declare two variables:

```
private ResourceDeployer<D, I> deployer;

private DeploymentValidator validator;
`````

...instead of a single one, even though the same object implements both interfaces. This usage makes it clear when the object is being used for deployment as opposed to deployment validation.